sdk/core/core/src/az_span_builder.c

```diff
@@ -22,3 +22,47 @@ AZ_NODISCARD az_result az_span_builder_append_byte(az_span_builder * const self,
   self->size += 1;
   return AZ_OK;
 }
+
+AZ_NODISCARD az_result az_span_builder_replace(
+    az_span_builder * const self,
+    size_t start,
+    size_t end,
+    az_span const span) {
+  AZ_CONTRACT_ARG_NOT_NULL(self);
+
+  size_t const current_size = self->size;
+  size_t const replaced_size = end - start;
+  size_t const size_after_replace = current_size - replaced_size + span.size;
+
+  // replaced size must be less or equal to current builder size. Can't replace more than what
+  // current is available
+  AZ_CONTRACT(replaced_size <= current_size, AZ_ERROR_ARG);
+  // start and end position must be before the end of current builder size
+  AZ_CONTRACT(start <= current_size && end <= current_size, AZ_ERROR_ARG);
+  // Start position must be less or equal than end position
+  AZ_CONTRACT(start <= end, AZ_ERROR_ARG);
+  // size after replacing must be less o equal than buffer size
+  AZ_CONTRACT(size_after_replace <= self->buffer.size, AZ_ERROR_ARG);
+
+  // For inserting, we just need append at the end
+  if (self->size == 0 || start == self->size) {
```

Comment:
nit: I don't think we need this optimization. `memove` should do nothing if src == dst.